### DISCUSSION TITLE
CssSyntaxError: postcss-import:  xxxxx/xxx/variables.scss:1:4: Unknown word

### DISCUSSION THREAD

**Discussion by sshallow** [tailwindlabs/tailwindcss#9430] [Help] [answered]
What version of Tailwind CSS are you using?

"tailwindcss": "^3.1.8"

What build tool (or framework if it abstracts the build tool) are you using?

```
{
  "name": "ruoyi",
  "scripts": {
    "dev": "vue-cli-service serve",
    "build:prod": "vue-cli-service build",
    "build:stage": "vue-cli-service build --mode staging",
    "preview": "node build/index.js --preview",
    "lint": "eslint --ext .js,.vue src"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "src/**/*.{js,vue}": [
      "eslint --fix",
      "git add"
    ]
  },
  "keywords": [
    "vue",
    "admin",
    "dashboard",
    "element-ui",
    "boilerplate",
    "admin-template",
    "management-system"
  ],
  "repository": {
    "type": "git",
    "url": "https://gitee.com/y_project/RuoYi-Vue.git"
  },
  "dependencies": {
    "@riophae/vue-treeselect": "0.4.0",
    "axios": "0.24.0",
    "clipboard": "2.0.8",
    "core-js": "3.19.1",
    "echarts": "4.9.0",
    "element-ui": "2.15.6",
    "file-saver": "2.0.5",
    "fuse.js": "6.4.3",
    "highlight.js": "9.18.5",
    "js-beautify": "1.13.0",
    "js-cookie": "3.0.1",
    "jsencrypt": "3.2.1",
    "nprogress": "0.2.0",
    "quill": "1.3.7",
    "screenfull": "5.0.2",
    "sortablejs": "1.10.2",
    "vue": "2.6.12",
    "vue-count-to": "1.0.13",
    "vue-cropper": "0.5.5",
    "vue-meta": "2.4.0",
    "vue-router": "3.4.9",
    "vuedraggable": "2.24.3",
    "vuex": "3.6.0"
  },
  "devDependencies": {
    "@vue/cli-plugin-babel": "4.4.6",
    "@vue/cli-plugin-eslint": "4.4.6",
    "@vue/cli-service": "4.4.6",
    "babel-eslint": "10.1.0",
    "babel-plugin-dynamic-import-node": "2.3.3",
    "chalk": "4.1.0",
    "compression-webpack-plugin": "5.0.2",
    "connect": "3.6.6",
    "eslint": "7.15.0",
    "eslint-plugin-vue": "7.2.0",
    "lint-staged": "10.5.3",
    "runjs": "4.4.2",
    "sass": "1.32.13",
    "sass-loader": "10.1.1",
    "script-ext-html-webpack-plugin": "2.1.5",
    "svg-sprite-loader": "5.1.1",
    "tailwindcss": "^3.1.8",
    "vue-template-compiler": "2.6.12"
  },
  "engines": {
    "node": ">=8.9",
    "npm": ">= 3.0.0"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions"
  ]
}

```
tailwind.config.js
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{html,js,vue,ts}"],
  theme: {
    extend: {},
  },
  plugins: [],
}
```
my index.css and variables.scss
![image](https://user-images.githubusercontent.com/21257340/192452499-f7fcbc09-838c-409c-b5c4-d9ebab9da9f1.png)
![image](https://user-images.githubusercontent.com/21257340/192452836-efc7f393-c112-40cc-9f56-54e458a267ec.png)

What browser are you using?

Chrome

What operating system are you using?

macOS

Describe your issue

I got an error when I followed the steps on the official website to the fourth step.
```
 npx tailwindcss -i ./src/assets/styles/index.scss -o ./dist/output.css --watch
CssSyntaxError: postcss-import: /Users/smart/Desktop/公司项目/fuba-stock/ruoyi-ui/src/assets/styles/variables.scss:1:4: Unknown word

> 1 | // base color
    |    ^
  2 | $blue:#324157;
  3 | $light-blue:#3A71A8;

```
I'm confused why he scans other css files(variables.scss) for errors
![image](https://user-images.githubusercontent.com/21257340/192452185-3a3173fd-1972-4e76-8e96-5a431a49a595.png)

**Comment by wongjn** [comment, marked as answer]
You have tried to import an Sass-language file into the Tailwind CLI, but CLI only interprets CSS. The error is the syntax error for double-slash (`//`) comments which exist in Sass but do not exist in CSS.

To integrate Tailwind with `@vue/cli-service`:
1. Add a `postcss.config.js` like this:  
   ```js
   module.exports = {
     plugins: {
       tailwindcss: {},
     },
   };
   ```
2. `import` your `index.css` somewhere in the "main" script file for your app. Looking at your project structure, this could be `src/App.vue` or `src/index.js` – you would add:
   ```js
   import './assets/styles/index.scss';
   ```
3. Run `npm run dev` to check it is working. The Vue CLI Service should build your Sass (and by extension, Tailwind) for you. You do not need to use the Tailwind CLI with the Vue CLI Service.

**Reply by sshallow** [reply]
PostCSS8 has compatibility issues with vue, and it has not been resolved so far, I have to give up tailwind3, thanks for your help

By the way, tailwind2 has a problem, when I use the following [config](https://tailwindcss.com/docs/configuration#important), the purpose is to only act on certain files. When other files use class="container", breakpoints will still be detected and take effect, but inline styles will not fail.

![image](https://user-images.githubusercontent.com/21257340/192466671-1f99da75-a4d6-4320-85bf-b03d2cb0cfea.png)

**Reply by wongjn** [reply]
The behavior you are experiencing works as designed, as per [the documentation](https://tailwindcss.com/docs/plugins#prefix-and-important-1):

> By default, component classes automatically respect the user’s `prefix` preference, but they are not affected by the user’s `important` preference.

This is relevant since the `container` class is a component class.
